File: Spotter/Spotter.Model/Responses/VenueResponse.cs
namespace Spotter.Model.Responses
{
    public class VenueResponse
    {
        public int Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public string Address { get; set; } = string.Empty;
        public int CityId { get; set; }
        public string CityName { get; set; } = string.Empty;
        public decimal? Latitude { get; set; }
        public decimal? Longitude { get; set; }
        public bool GeocodingPending { get; set; }
    }
}

File: UI/spotter_mobile/lib/core/models/venue_response.dart
class VenueResponse {
  final int id;
  final String name;
  final String address;
  final int cityId;
  final String cityName;
  final double? latitude;
  final double? longitude;
  final bool geocodingPending;

  VenueResponse({
    required this.id,
    required this.name,
    required this.address,
    required this.cityId,
    required this.cityName,
    this.latitude,
    this.longitude,
    required this.geocodingPending,
  });

  factory VenueResponse.fromJson(Map<String, dynamic> json) {
    return VenueResponse(
      id: json['id'] as int,
      name: json['name'] as String,
      address: json['address'] as String,
      cityId: json['cityId'] as int,
      cityName: json['cityName'] as String,
      latitude: json['latitude'] != null ? (json['latitude'] as num).toDouble() : null,
      longitude: json['longitude'] != null ? (json['longitude'] as num).toDouble() : null,
      geocodingPending: json['geocodingPending'] as bool? ?? false,
    );
  }
}
